### PR TITLE
feat: add template for local pubsub using java sdk

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
Add a new template in java as example for local pubsub communication between components. 

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/java/LocalPubSub/pom.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  <packaging>jar</packaging>
+  <version>1.0.0</version>
+  <name>LocalPubSub</name>
+  <url>http://maven.apache.org</url>
+  <properties>
+    <junitVersion>5.5.2</junitVersion>
+    <mavenPluginVersion>2.1</mavenPluginVersion>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <iotDeviceSdkVersion>1.10.6</iotDeviceSdkVersion>
```
Comment (MikeDombo) [inline templates/java/LocalPubSub/pom.xml:16]: use the latest version.

File: templates/java/LocalPubSub/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            - aws.greengrass#PublishToTopic
+          resources:
+            - "/topic/local/pubsub"
+Manifests:
+  - Platform:
+      os: all
+    Artifacts:
+      - URI: "s3://BUCKET_NAME/COMPONENT_NAME/COMPONENT_VERSION/LocalPubSub-1.0.0.jar"
+    Lifecycle:
+      Run: "java -cp {artifacts:path}/LocalPubSub-1.0.0.jar com.localpubsub.App {configuration:/Topic} {configuration:/Message}"
```
Comment (MikeDombo) [inline templates/java/LocalPubSub/recipe.yaml:26]: put topic and message within quotes

File: templates/java/LocalPubSub/src/main/java/com/localpubsub/App.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
+
+import java.util.Arrays;
+
+public class App {
+
+    public static void main(String[] args) {
+        String topic = args[0];
+        String message = String.join(" ", Arrays.copyOfRange(args, 1, args.length));
+        try (GreengrassCoreIPCClientV2 ipcClient = GreengrassCoreIPCClientV2.builder().build()) {
```
Comment (MikeDombo) [inline templates/java/LocalPubSub/src/main/java/com/localpubsub/App.java:15]: while technically correct, I'm not sure if this sets our customers up for success. I'd recommend putting the client as a field of the class and initializing it when starting up. Do not close it.

File: templates/java/LocalPubSub/src/main/java/com/localpubsub/App.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCClientV2;
+import software.amazon.awssdk.aws.greengrass.SubscribeToTopicResponseHandler;
+import software.amazon.awssdk.aws.greengrass.model.SubscribeToTopicResponse;
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public class App {
+    private  final GreengrassCoreIPCClientV2 ipcClient ;
```
Comment (MikeDombo) [inline templates/java/LocalPubSub/src/main/java/com/localpubsub/App.java:12]: just make it static?

File: templates/java/LocalPubSub/src/main/java/com/localpubsub/App.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public class App {
+    private static GreengrassCoreIPCClientV2 ipcClient ;
+
+    public App() throws IOException {
+        ipcClient = GreengrassCoreIPCClientV2.builder().build();
+    }
```
Comment (MikeDombo) [inline templates/java/LocalPubSub/src/main/java/com/localpubsub/App.java:16]: now get rid of this and just initialize it in main.

**Review by MikeDombo — APPROVED** [review]

**Review by Nelsonochoam — APPROVED** [review]

**Pull request merged**